File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> storage(nums.begin(), nums.end());

        int result = 0;
        for (auto num : storage){
            int len = 1;
            if (storage.find(num-1) == storage.end()){
                int cur = num;
                while (storage.find(cur+1) != storage.end()){
                    len++;
                    cur++;
                }
            }
            result = max(result, len);
        }
        return result;

}

};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        storage = dict()

        for num in nums:
            if num not in storage:
                storage[num] = 1
        
        ans = 0
        for num in storage:
            cur_len = 1
            cur_num = num + 1
            if num-1 not in storage:
                while cur_num in storage:
                    cur_len += 1
                    cur_num += 1
            ans = max(ans, cur_len)
        return ans